### PR TITLE
Bug 1462758 - upgrade queue to use newest libraries, rootUrl

### PR DESCRIPTION
**Pull request by djmitche**
I'm not totally happy with this, mostly because the "real" tests remain a bit intermittent.  I'm tempted to do some refactoring to fix that, but this is already a big change, and I don't think it makes sense to add risk (by refactoring) just to make tests better.  At least, not right now.

@jonasfj could you do a review as a parting gift? :)

### CODE REVIEW

**Comment by djmitche** [comment]
This will depend on https://github.com/taskcluster/taskcluster-lib-testing/pull/30 and, I think, https://github.com/taskcluster/taskcluster-lib-testing/pull/29

**Comment by djmitche** [comment]
The tc-lib-testing changes are now landed.

**Comment by djmitche** [comment]
GREEEEEEEN!

File: src/workerinfo.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ class WorkerInfo {
           });
         }
 
-        createEntry(this.Provisioner, {
+        return createEntry(this.Provisioner, {
```
Comment (imbstack) [inline src/workerinfo.js]: This makes the tests much nicer!

File: schemas/v1/worker-response.yml [hunk 2]
```diff
@@ -42,7 +42,7 @@ properties:
       List of 20 most recent tasks claimed by the worker.
     type:         array
     items:
-      $ref:       "http://schemas.taskcluster.net/queue/v1/task-run.json#"
+      $ref:       "task-run.json#"
```
[the remark this answers is not in the document]
Comment (djmitche) [reply schemas/v1/worker-response.yml]: This makes a bit more sense now (even if the implementation is a little weird) -- in the API declaration, we refer to the files (implementation referring to implementation).  Everywhere else, we refer to the schema (schemas referring to schemas).  But yeah, still a little odd.

File: src/workerinfo.js (hunk 1 above)
Comment (djmitche) [reply src/workerinfo.js]: And fixes a pretty major bug :)

**Pull request merged**